src/ContextFreeTasks.Shared/ContextFreeTask.cs

```
@@ -1,13 +1,15 @@
 ﻿using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
+using static System.Threading.Tasks.Task;
 
 namespace ContextFreeTasks
 {
     [AsyncMethodBuilder(typeof(AsyncContextFreeTaskMethodBuilder))]
     public struct ContextFreeTask
     {
-        public Task Task { get; }
-        internal ContextFreeTask(Task t) => Task = t;
+        private readonly Task _task;
+        public Task Task => _task ?? FromResult(default(object));
+        internal ContextFreeTask(Task t) => _task = t;
```

So far, `await` on `null` should throw an Exception. For the purpose of null-safety, the constructor should throw an exception if `t` is `null` and `_task` should not be `null`. Of course, `default(ContextFreeTask)` makes `_task` be `null`, but, it is misuse.

I hope the following proposals:

https://github.com/dotnet/roslyn/issues/7171
https://github.com/dotnet/roslyn/issues/15108

---

Yes, I concerned about `default(ContextFreeTask)`. if provides struct as public, we must consider the default.
for example, `ValueTask` created with default keyword, it is same as `FromResult(default(T))`.
the constructor of `ContextFreeTask` is hid to internal. so ONLY `default(ContextFreeTask)` makes `_task` be `null`. it's very rare case.